### DISCUSSION TITLE
How would you integrate this into a unity package?

### DISCUSSION THREAD

**Discussion by codewing** [getsentry/sentry-unity#1144] [Q&A] [answered]
We're building a upm package for other projects and I've added this package as a dependency and initialized it with the `SentrySdk.Init()` call.
If the unity app that includes us would also like to use sentry there isn't a way they can or am I missing something?

e.g. calling SentrySdk.CaptureMessage() would cause them to send the message to our backend?

**Comment by mattjohnsonpint** [comment, marked as answer]
Hi.  Sentry is designed for applications, not for libraries.  You can add a Sentry SDK to your samples and/or test applications, but we do not support adding it to a component or library you are developing for integration into other applications.

There are technical reasons why, but the primary reason is that the privacy agreement for the data that the Sentry SDK collects is between Sentry and the owner of the application only, and doesn't involve third parties.

Some previous discussions about this here:  https://github.com/getsentry/sentry-dotnet/issues/1872#issuecomment-1251194629

Thanks.
